src/analysis/bounds.rs

```diff
@@ -119,15 +121,19 @@ impl Bounds {
         func: &Function,
         par: &CParameter,
         async: bool,
+        expecting_data: bool,
     ) -> (Option<String>, Option<CallbackInfo>) {
         let type_name = bounds_rust_type(env, par.typ);
-        let mut type_string = if async && async_param_to_remove(&par.name) {
+        let mut type_string = if (async && async_param_to_remove(&par.name)) ||
+                                 (expecting_data && (par.name == "data" ||
+                                  par.name.ends_with("_data"))) {
```

Comment:
Matching on names of parameters is not a good idea, it will lead to breakages along the way. Match on types if there's nothing else, but ideally use the annotations from the GIR metadata

Reply:
This is a fix for the async generation but I agree.